ionoscloud/resource_autoscaling_group.go

```diff
@@ -0,0 +1,483 @@
+)
+
+func resourceAutoscalingGroup() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceAutoscalingGroupCreate,
+		ReadContext:   resourceAutoscalingGroupRead,
+		UpdateContext: resourceAutoscalingGroupUpdate,
+		DeleteContext: resourceAutoscalingGroupDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceAutoscalingGroupImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"max_replica_count": {
+				Type:             schema.TypeInt,
+				Description:      "The maximum value for the number of replicas on a VM Auto Scaling Group. Must be >= 0 and <= 200. Will be enforced for both automatic and manual changes.",
+				Required:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.IntBetween(0, 200)),
+			},
+			"min_replica_count": {
+				Type:             schema.TypeInt,
+				Description:      "The minimum value for the number of replicas on a VM Auto Scaling Group. Must be >= 0 and <= 200. Will be enforced for both automatic and manual changes",
+				Required:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.IntBetween(0, 200)),
+			},
+			//"target_replica_count": {
+			//	Type:             schema.TypeInt,
+			//	Description:      "The target number of VMs in this Group. Depending on the scaling policy, this number will be adjusted automatically. VMs will be created or destroyed automatically in order to adjust the actual number of VMs to this number. If targetReplicaCount is given in the request body then it must be >= minReplicaCount and <= maxReplicaCount.",
+			//	Optional:         true,
+			//	Computed:         true,
+			//	ValidateDiagFunc: validation.ToDiagFunc(validation.IntBetween(0, 200)),
+			//},
+			"name": {
+				Type:             schema.TypeString,
+				Description:      "User-defined name for the Autoscaling Group.",
+				Required:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.StringLenBetween(1, 255)),
+			},
+			"policy": {
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Description: "Defines the behavior of this VM Auto Scaling Group. A policy consists of triggers and actions, where an action is an automated behavior, and the trigger defines the circumstances under which the action is triggered. Currently, two separate actions are supported, namely scaling inward and outward, triggered by the thresholds defined for a particular metric.",
+				Required:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"metric": {
+							Required:         true,
+							Type:             schema.TypeString,
+							Description:      "The Metric that should trigger the scaling actions. Metric values are checked at fixed intervals.",
+							ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice([]string{"INSTANCE_CPU_UTILIZATION_AVERAGE", "INSTANCE_NETWORK_IN_BYTES", "INSTANCE_NETWORK_IN_PACKETS", "INSTANCE_NETWORK_OUT_BYTES", "INSTANCE_NETWORK_OUT_PACKETS"}, true)),
```

Comment:
@cristiGuranIonos I see that we are using these strings in multiple places so maybe we can define them as constants.

Reply:
they can be taken from sdk actually